bazel-build.el

```diff
@@ -26,24 +26,60 @@
 
 (defun bazel-build (target)
   "Build a Bazel TARGET."
-  (interactive "sbazel build ")
+  (interactive (bazel-build--read-target "bazel build "))
   (bazel-build--run-bazel-command "build" target))
 
 (defun bazel-run (target)
   "Build and run a Bazel TARGET."
-  (interactive "sbazel run ")
+  (interactive (bazel-build--read-target "bazel run "))
   (bazel-build--run-bazel-command "run" target))
 
 (defun bazel-test (target)
   "Build and run a Bazel test TARGET."
-  (interactive "sbazel test ")
+  (interactive (bazel-build--read-target "bazel test "))
   (bazel-build--run-bazel-command "test" target))
 
 (defun bazel-build--run-bazel-command (command target)
   "Run Bazel tool with given COMMAND, e.g. build or run, on the given TARGET."
   (compile
    (mapconcat #'shell-quote-argument (list "bazel" command target) " ")))
 
+(defun bazel-build--read-target (prompt)
+  "Read a Bazel build target from the minibuffer.  PROMPT is a read-only prompt."
+  (let* ((file-name (buffer-file-name))
+         (workspace-root (bazel-build--find-workspace-root file-name))
+         (initial-input
+          (bazel-build--suggest-initial-input file-name workspace-root)))
+  (read-from-minibuffer prompt initial-input)))
+
+(defun bazel-build--find-workspace-root (file-name)
+  "Find the root of the Bazel workspace containing FILE-NAME."
+  (let ((workspace-root (locate-dominating-file file-name "WORKSPACE")))
+    (if workspace-root (expand-file-name workspace-root)
+      (error "Not in a Bazel workspace"))))
+
+(defun bazel-build--suggest-initial-input (file-name workspace-root)
+  "Suggest initial input to the interactive completion.
+FILE-NAME is the file-name of the current buffer.  WORKSPACE-ROOT is
+the root of the Bazel workspace."
+  (concat "//" (directory-file-name (bazel-build--extract-package-name
+                                     file-name workspace-root))))
+
+(defconst bazel-build--recognized-build-file-names
+  '("BUILD.bazel" "BUILD")
```

Comment:
I'd inline this constant since it's only used once.

Reply:
Done.